### ISSUE TITLE
Windows, clojure client: Log and HUD has inconsistent line endings 

### ISSUE THREAD

**Issue by maksut** [Olical/conjure#446]
Possibly similar to: https://github.com/Olical/conjure/issues/439

Clojure client
Windows  10
NVIM v0.8.1

Log and HUD renders ^M for some evaluations and doc strings (via K). I believe this is Vim way of saying that the buffer has inconsistent line endings.

Evaluating `(println "foo bar")` prints:
```
; eval (current-form): (println "foo bar")
; (out) foo bar^M
nil
```

K on `println` prints:
```
; doc (word): println
; -------------------------^M
; clojure.core/println^M
; ([& more])^M
;   Same as print followed by (newline)^M
```

Updating these two functions in `fnl\conjure\text.fnl` seems working for me:
```
(defn trailing-newline? [s]
  (string.match s "\r?\n$"))

(defn trim-last-newline [s]
  (string.gsub s "\r?\n$" ""))
```

Sorry, I can't put a PR together because windows dev environment for conjure is a bit crippled.

**Comment by Olical** [comment]
Ah yes, this will definitely be that issue. I'll just have to change every client to handle both kinds of line endings really, will get to that soon. I don't have a Windows dev environment and have never tried to use Neovim in Windows, not sure what needs to happen to get the dev scripts working there I'm afraid.

**Comment by Olical** [comment]
I've added this to develop, let me know if it fixes all of the issues for you! And thank you for sharing a potential fix! Seems okay to me but I'm not on Windows so I haven't tried to repro. It at least doesn't break the tests or Linux usage.

**Comment by maksut** [comment]
Just tried it. Works as expected. Thanks a lot!

Conjure looks great and Neovim feels much better than last time I've tried a few years ago.

Dev environment troubles was mostly because me figuring out things. I don't have `bash` in this setup so I had to figure out how to compile fnl files of the plugin with aniseed. I've ended up running this manually:
```
lua require('aniseed.compile').glob('**/*.fnl', 'fnl', 'lua')
```

Again thanks for your work. Conjure brings joy.
